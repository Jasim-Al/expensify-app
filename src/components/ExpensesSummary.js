import React from 'react';
import { connect } from 'react-redux';
import expensesSelector from '../selectors/expenses';
import expensesSum from '../selectors/expenses-total';
import numeral from 'numeral'

export const ExpensesSummary = props => (
    <div>
       <h1>Viewing {props.expensesLength} {props.expensesLength > 1 ? 'expenses' : 'expense' } totalling  { props.expensesLength !== 0 ? numeral(props.sum/100).format('$0,0.00') : '0' } </h1>
    </div>
);

const mapStateToProps = state => {
    return {
        expensesLength : expensesSelector(state.expenses, state.filters).length,
        sum:expensesSum(state.expenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);