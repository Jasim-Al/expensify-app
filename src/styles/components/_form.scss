.form {
    display: flex;
    flex-direction: column;
    div{
        div{
            width: 100%;
        }
    }
    > * {
        margin-bottom: $m-size;
    }
}

.form__error {
    margin: 0 0 $m-size 0;
    font-style: italic;
}