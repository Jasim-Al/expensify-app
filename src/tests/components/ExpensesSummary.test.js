import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('render ExpensesSummary with one expense message', () => {
    let wrapper = shallow( <ExpensesSummary expensesLength = { 1 } sum = { 70 } /> );
    expect(wrapper).toMatchSnapshot();
});

test('render ExpensesSummary with multiple expenses message', () => {
    let wrapper = shallow( <ExpensesSummary expensesLength = { 3 } sum = { 390 } /> );
    expect(wrapper).toMatchSnapshot();
});