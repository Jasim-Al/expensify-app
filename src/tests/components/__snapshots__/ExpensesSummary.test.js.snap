// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ExpensesSummary with multiple expenses message 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling  
    $3.90
     
  </h1>
</div>
`;

exports[`render ExpensesSummary with one expense message 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling  
    $0.70
     
  </h1>
</div>
`;
